azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/presentation/fragment/setup/components/SetupControlBarView.kt

```diff
@@ -201,17 +183,17 @@ internal class SetupControlBarView : LinearLayout {
 
     private fun toggleAudio() {
         if (micButton.isON) {
-            viewModel.turnMicOff()
+            viewModel.turnMicOff(context)
```

Comment:
This business logic should be pushed up to the VM I think.

I.e.
instead of 
        micButton.setOnClickListener {
            toggleAudio()
        }
       
it should be
micButton.setOnClickListener(viewModel::onMicClicked)

And the logic about what to do from there (i.e. mic on/off etc, should live in the VM).

i.e.

onMicClicked(context) {
  dispatchCorrectActionForMicNow()
  notifyMicClickedListener()
}

Reply:
I think this is the actual button state logic that should stay in the view.
ViewModel should not keep that state, but putting this state on Redux is a bit too much.
It was this way already.